api/objects.proto

```diff
@@ -222,3 +222,35 @@ message Cluster {
 	// a new key is allocated on key rotation.
 	uint64 encryption_key_lamport_clock = 6;
 }
+
+// Secret represents a secret that should be passed to a container or a node,
+// and is immutable. It wraps the `spec` provided by the user with useful
+// information that is generated from the secret data in the `spec`, such as
+// the digest and size of the secret data.
+message Secret {
```

Comment:
It would be possible to write the code to avoid that. A Raft write transaction has an exclusive lock on the Raft store, so if the listing and calculation is done inside the write transaction, a unique version ID can be guaranteed.